src/mvc/collection.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Collections are ordered sets of models.
+ *
+ * Any event that is triggered on a model in a collection will also be triggered on the collection directly, for
+ * convenience.
+ *
+ * @class Collection
+ */
+
+CKEDITOR.define( [ 'emitter', 'utils' ], function( EmitterMixin, utils ) {
+	function Collection() {
+		/**
+		 * The internal list of models in the collection.
+		 *
+		 * @property _models
+		 * @private
+		 */
+		Object.defineProperty( this, '_models', {
+			value: []
+		} );
+
+		/**
+		 * The number of items available in the collection.
+		 *
+		 * @property length
+		 */
+		Object.defineProperty( this, 'length', {
+			get: function() {
+				return this._models.length;
+			}
+		} );
+	}
+
+	/**
+	 * @inheritdoc utils#extend
+	 */
+	Collection.extend = utils.extendMixin;
+
+	utils.extend( Collection.prototype, EmitterMixin, {
+		/**
+		 * Adds an item into the collection.
+		 *
+		 * Note that this is an array-like collection, so the same item can be present more than once. This behavior is
+		 * for performance purposes only and is not guaranteed to be kept in the same way in the future.
+		 *
+		 * @param {Model} model The item to be added.
+		 */
+		add: function( model ) {
+			model.addParentEmitter( this );
```

Comment:
I would go without the event bubbling. I think that it's hard now to say whether we need it and how we need it (and our case isn't so typical). If, based on the cases on which we'll work, we'll see that we often have to [forward events](https://github.com/ckeditor/ckeditor5-design/issues/7#issuecomment-74501808) we can make bubbling possible in some way. We'll also see if there are name conflicts, any performance drop, etc. But if we'll go with bubbling now I feel that it will be harder to evaluate and undo it in the future.

In other words - KISS for now and let's reevaluate this with time. Such problems are best handled iteratively.